### PR TITLE
Clang Support

### PR DESCRIPTION
**Pull request by onthegrid007**
Adds Clang support
Fixes issues:
https://github.com/luxonis/XLink/issues/80
https://github.com/luxonis/depthai-core/issues/1044

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -92,7 +92,10 @@ if(XLINK_ENABLE_LIBUSB)
         if(WIN32 AND NOT MINGW)
```
Comment (themarpe) [inline CMakeLists.txt:92]:
```suggestion
        if(WIN32 AND CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
```

Comment (themarpe) [reply CMakeLists.txt:92]: Probably above is the way to go then, as it seems that DELAYLOAD is MSVC feature (and not "everyone's but Clang's feature)
Also, the delayimp.lib then isn't required.

**Review by themarpe — CHANGES_REQUESTED** [review]

**Comment by onthegrid007** [comment]
Could this get pushed downstream so depthai-core can use it properly?

File: CMakeLists.txt [hunk 2]
```diff
@@ -89,10 +89,13 @@ if(XLINK_ENABLE_LIBUSB)
...
         target_link_libraries(${TARGET_NAME} PRIVATE usb-1.0)
-        if(WIN32 AND NOT MINGW)
+        if(WIN32 AND CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
             target_link_libraries(${TARGET_NAME} PRIVATE delayimp.lib)
             # workaround https://gitlab.kitware.com/cmake/cmake/-/issues/20022
-            target_link_options(${TARGET_NAME} PUBLIC "$<LINK_ONLY:/DELAYLOAD:libusb-1.0$<$<CONFIG:Debug>:d>.dll>")
+			if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+				# fix to https://github.com/luxonis/depthai-core/issues/1044
+				target_link_options(${TARGET_NAME} PUBLIC "$<LINK_ONLY:/DELAYLOAD:libusb-1.0$<$<CONFIG:Debug>:d>.dll>")
+			endif()
```
Comment (themarpe) [inline CMakeLists.txt:98]: Seems to be too indented & with tabs instead of space?

Comment (themarpe) [reply CMakeLists.txt:98]: Also, this if&endif statement can be removed, given that above if only allows msvc

**Review by themarpe — CHANGES_REQUESTED** [review]

**Comment by onthegrid007** [comment]
Done.

**Review by themarpe — CHANGES_REQUESTED** [review]
@onthegrid007 sorry, one additional thing - do you mind adding it to CI as well?

**Comment by themarpe** [comment]
(otherwise LGTM)

**Comment by onthegrid007** [comment]
I can if you could tell me what the CI is...?

**Comment by themarpe** [comment]
@onthegrid007 the Continious Integration - there is a .yml workflow file that runs "tests/builds" of the codebase using various platforms/operating systems. `windows-latest` currently only does MSVC, so we should add also Clang there

**Comment by onthegrid007** [comment]
Yes I can try. Give ma little bit

**Comment by onthegrid007** [comment]
Done.

**Comment by onthegrid007** [comment]
It succeeded...

**Comment by themarpe** [comment]
I mean that clang wasn't used to compile, but it was still MSVC: 

> -- The CXX compiler identification is MSVC 19.40.33812.0

Under the windows, clang build

**Comment by onthegrid007** [comment]
I'll make a change and see if it fixes it

**Comment by onthegrid007** [comment]
Ninja or some other makefile system is required for cmake to execute using clang properly, that has been added and should allow the CI to propagate...

**Review by themarpe — APPROVED** [review]
CC: @moratom please integrate, thanks

**Pull request merged**